### DISCUSSION TITLE
How to use GitPython with GitHub Personal Access Token for Authentication inside a GitHub Action

### DISCUSSION THREAD

**Discussion by matroscoe** [gitpython-developers/GitPython#1580] [Q&A]
Trying to use GitPython to do some automated stuff during a GitHub action, GitHub being what it is requires the use of a PAT in order to access a repo to do a push when the branches are protected. 

How do I clone and push using GitPython when I have to provide a PAT. It was working using the `https` URL but GitHub recently killed that and I am not sure how to enforce that GitPython uses the PAT for authentication against GitHub.

**Comment by Byron** [comment]
Is it possible to set the PAT using [http.extraHeader](https://git-scm.com/docs/git-config#Documentation/git-config.txt-httpextraHeader) using `Authorization: Basic base64(user:password)`? 

If so, maybe it's also possible to transiently set it using the [GIT_CONFIG_COUNT…](https://git-scm.com/docs/git-config#ENVIRONMENT) environment variables.
